libconfluo/confluo/atomic_multilog.h

```diff
@@ -245,6 +247,13 @@ class atomic_multilog {
    */
   size_t append(void *data);
 
+  /**
+   * Appends json-formatted data to the atomic multilog
+   * @param json_data The json-formatted data to be stored
+   * @return The offset of where the data is located
+   */
+  size_t append_json(void *json_data);
```

Comment:
It would be better if the argument to the function is of type `std::string`, since the first thing you do in the function is to write the `void*` argument into a `std::stringstream` anyway. `void*` arguments are prone to misuse (type-safety), and shouldn't be used without very good reason.